### ISSUE TITLE
[ONNX] torch.flatten for 1D tensor results in wrong ONNX

### ISSUE THREAD

**Issue by shinh** [pytorch/pytorch#67670]
## 🐛 Bug

It seems if you export `torch.flatten(x)` where `x.ndim == 1`, torch.onnx exports a single `Flatten` op whose output is always 2D.

## To Reproduce

```
class Flatten(torch.nn.Module):
    def forward(self, x):
        return x.flatten()


x = torch.rand(100)
torch.onnx.export(Flatten(), x, 'flatten.onnx', opset_version=13)

print(Flatten()(x).shape)
print(onnx.load("flatten.onnx").graph.output[0])
```

You will see

```
torch.Size([100])
name: "1"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_value: 1
      }
      dim {
        dim_value: 100
      }
    }
  }
}
```

which means torch expects 1D output but ONNX outputs 2D.

I think [this condition](https://github.com/pytorch/pytorch/blob/cd51d2a3ecc8ac579bee910f6bafe41a4c41ca80/torch/onnx/symbolic_opset11.py#L798) is wrong. For 1D tensor with default parameters, `dim=1`, `start_dim=0`, and `end_dim=-1`. This condition becomes unexpectedly `True` as `end_dim == dim - 2`. I think we should skip this if `end_dim` is negative, or normalize negative `end_dim`.

## Expected behavior

Flatten should not be used when the result will be 1D tensor. The exporter should call `sym_help._flatten_helper`.

## Environment

```
Collecting environment information...
PyTorch version: 1.9.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.2 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: 8.0.1-9 (tags/RELEASE_801/final)
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.8.10 (default, Jun  2 2021, 10:49:15)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.4.0-72-generic-x86_64-with-glibc2.29
Is CUDA available: True
CUDA runtime version: 11.3.109
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1060 3GB
Nvidia driver version: 465.19.01
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.2.1
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] mypy==0.812
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.19.3
[pip3] pytorch-pfn-extras==0.5.1
[pip3] pytorchcv==0.0.65
[pip3] torch==1.9.0
[pip3] torch-dftd==0.3.0
[pip3] torchvision==0.10.0
[conda] Could not collect
```

## Additional context

I only tries torch 1.9.0, but I think the problem is still there https://github.com/pytorch/pytorch/blob/cd51d2a3ecc8ac579bee910f6bafe41a4c41ca80/torch/onnx/symbolic_opset11.py#L798

**Comment by fatcat-z** [comment]
Traced by https://msdata.visualstudio.com/Vienna/_workitems/edit/1502114.

**Comment by BowenBao** [comment]
Fixed in latest PyTorch. Closing.

```
torch.Size([100])
name: "1"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_value: 100
      }
    }
  }
}
```
